guidelines.md

```diff
@@ -0,0 +1,91 @@
+``` $ git config --global user.name "YOUR NAME" ```
+```$ git config --global user.email "YOUR EMAIL ADDRESS"```
+This is an important step to mark your commits to your name and email.
+
+### FORK A PROJECT -
+***
+You can use github explore - https://github.com/explore to find a project that interests you and match your skills. Once you find your cool project to workon, you can make a copy of project to your account. This process is called forking a project to your Github account. On Upper right side of project page on Github, you can see -
+
+<p align="center">  <img  src="https://i.imgur.com/P0n6f97.png">  </p>
+
+Click on fork to create a copy of project to your account. This creates a separate copy for you to workon.
+
+### FINDING A FEATURE OR BUG TO WORKON - 
+***
+Open Source projects always have something to workon and improves with each new release. You can see the issues section to find something you can solve or report a bug. The project managers always welcome new contributors and can guide you to solve the problem. You can find issues in the right section of project page.
+
+<p align="center">  <img  src="https://i.imgur.com/czVjpS7.png">  </p>
+
+### CLONE THE FORKED PROJECT -
+***
+You have forked the project you want to contribute to your github account. To get this project on your development machine we use clone command of git.
+
+```$ git clone https://github.com/<your-account-username>/<your-forked-project>.git```
+Now you have the project on your local machine.
+
+### ADD A REMOTE (UPSTREAM) TO ORIGINAL PROJECT REPOSITORY 
+***
+Remote means the remote location of project on Github. By cloning, we have a remote called origin which points to your forked repository. Now we will add a remote to the original repository from where we had forked.
+
+```$ cd <your-forked-project-folder>```
+```$ git remote add upstream https://github.com/<author-account-username>/<project>.git```
+You will see the benefits of adding remote later.
+
+### SYNCHRONIZING YOUR FORK -
+***
+Open Source projects have a number of contributors who can push code anytime. So it is necessary to make your forked copy equal with the original repository. The remote added above called Upstream helps in this.
+
+```$ git checkout master```
+```$ git fetch upstream```
+```$ git merge upstream/master```
+```$ git push origin master```
+The last command pushes the latest code to your forked repository on Github. The origin is the remote pointing to your forked repository on github.
+
+### CREATE A NEW BRANCH FOR A FEATURE OR BUGFIX -
+***
+Normally, all repositories have a master branch which is considered to remain stable and all new features should be made in a separate branch and after completion merged into master branch. So we should create a new branch for our feature or bugfix and start working on the issue.
+
+```$ git checkout -b <feature-branch>```
+This will create a new branch out of master branch. Now start working on the problem and commit your changes.
```

Comment:
This sentence continuous in the same line as the prev git command (when rendering). Start this in a new line.